elastic/kibana
# [ObsUX] [Infra] Add Infra locators

Closes https://github.com/elastic/kibana/issues/178520

#### What was done
- add locator for asset details page
- add locator for asset details flyout
- add locator for hosts page
- tests to cover the new locators


#### How to test
What I did to test this is to change the link in https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/infra/public/components/asset_details/links/link_to_node_details.tsx

i.e for hosts locator
Step 1
Get the locator
```
  const { services } = useKibanaContextForPlugin();
  const hostsLocator = services.observabilityShared.locators.infra.hostsLocator;
```
Step 2
Set locator params
```
const hostsUrl = hostsLocator?.getRedirectUrl({
    query: {
      language: 'kuery',
      query: `host.name: ${host.name}`,
    },
    dateRange,
    limit: 50,
  });
```
Step 3
Pass the url to the button
```
 <EuiButtonEmpty
      data-test-subj="infraAssetDetailsOpenAsPageButton"
      size="xs"
      flush="both"
      href={hostsUrl}
      >
 ```
 Step 4
 In Hosts view page, open a host flyout and click on `Open as page`, should redirect with the correct url set with the locator
